interface/views.py

```diff
@@ -74,6 +75,26 @@ def homepage(request):
                    'logout_url': redirect(logout_view).url})
 
 
+@login_required
+def review(request, pk):
+    if request.POST and request.user.is_staff:
+        marks = re.findall(
+            r'^([+-]\d\.\d):',
+            request.POST['review-code'],
+            re.MULTILINE
+        )
+        log.debug('Marks found: ' + str(marks))
+
+        review_score = sum([decimal.Decimal(mark) for mark in marks])
+
+        submission = get_object_or_404(models.Submission, pk=pk)
```

Comment:
You should do this at the beginning of the function, if `pk` is invalid then 404 is the only answer.